modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -71,10 +68,6 @@
 import static org.elasticsearch.index.reindex.remote.RemoteResponseParsers.RESPONSE_PARSER;
 
 public class RemoteScrollableHitSource extends ScrollableHitSource {
-    /**
-     * The maximum size of the remote response to buffer. 200mb because bulks beyond 40mb tend to be slow anyway but 200mb is simply huge.
-     */
-    private static final ByteSizeValue BUFFER_LIMIT = new ByteSizeValue(200, ByteSizeUnit.MB);
```

Comment:
I just picked 200MB because 100MB seemed like a "very big buffer" for reindexes and 200MB was sure to contain it. Given our default heap size of 1GB that gives plenty of room for a small node to do this.

Maybe we should make this value more visible on the client builder. That way people think about it. Most folks are probably fine with a 10mb or 20mb limit, but reindexing really is a bigger slices thing. Pretty much any scrolling is going to be like that.